### PR TITLE
SI-6580 Scaladoc: Should not close void elements

### PR DESCRIPTION
**Pull request by kzys**
Because it will generate a useless element like "</img>".

To made matters worse, Scaladoc used to generate the element with
attributes (like </img src="...">). That's why we had SI-6580.

### CODE REVIEW

**Comment by JamesIry** [comment]
Please request a reviewer.

**Comment by kzys** [comment]
@VladUreche @heathermiller Could you take a look?

File: src/compiler/scala/tools/nsc/doc/base/comment/Body.scala [hunk 1]
```diff
@@ -75,16 +75,27 @@ object EntityLink {
...
+  def close =
+    tagName match {
+      case None =>
+        None
+      case Some("br" | "img") =>
+        None
+      case Some(name) =>
+        Some(HtmlTag("</" + name + ">"))
+    }
 }
```
Comment (paulp) [inline src/compiler/scala/tools/nsc/doc/base/comment/Body.scala]:
```
def close = tagName collect { case name if name != "br" && name != "img" => HtmlTag(s"</$name>") }
```

Or probably better,

```
private val TagsNotToClose = Set("br", "img")
def close = tagName collect { case name if !TagsNotToClose(name) => HtmlTag(s"</$name>") }
```

Comment (paulp) [inline src/compiler/scala/tools/nsc/doc/base/comment/Body.scala]: A few "Option" things which invariably mean there is a better expression available:
- case None => None 
- case Some(x) => x
- getOrElse true, getOrElse false, getOrElse null
- .get

**Comment by kzys** [comment]
Thanks for reviewing. I updated the branch.

**Comment by paulp** [comment]
LGTM.

**Pull request merged**